src/apprt/embedded.zig

```diff
@@ -1550,6 +1550,11 @@ pub const CAPI = struct {
         return selection.len;
     }
 
+    /// Clear the current selection in the surface.
+    export fn ghostty_surface_clear_selection(surface: *Surface) void {
+        surface.core_surface.io.terminal.screen.clearSelection();
```

Comment:
This is unsafe because the IO thread owns the terminal, not the GUI thread. This requires a lock. But the lock shouldn't be held outside of Surface.zig. We should add a `clearSelection` function to `Surface.zig`.